lib/Templating/TwigDefaultDelegatingEngine.php

```diff
@@ -100,14 +95,38 @@ public function isDelegate()
         return $this->delegate;
     }
 
-    /**
-     * @return Environment
-     */
-    public function getTwigEnvironment(): Environment
+    public function getTwigEnvironment(bool $sandboxed = false): Environment
     {
+        if ($sandboxed) {
+            if (!$this->twig->hasExtension(SandboxExtension::class)) {
+                $securityPolicy = $this->config['templating_engine']['twig']['sandbox_security_policy'];
+
+                $tags = $securityPolicy['tags'];
+                $filters = $securityPolicy['filters'];
+                $functions = $securityPolicy['functions'];
+
+                $policy = new SecurityPolicy($tags, $filters, $functions);
+                $sandbox = new SandboxExtension($policy);
+                $this->twig->addExtension($sandbox);
```

Comment:
I think this dynamic approach could cause some issues in case the Twig environment was already initialized previously, then `addExtension()` will throw an exception. 
I'd propose to register `SandboxExtension` as a service here: https://github.com/pimcore/pimcore/blob/3a539e605e5bf4f4924c83366837c0b840933315/bundles/CoreBundle/config/templating_twig.yaml#L76 

It will the get automatically registered as a Twig extension and here, we can just use `enableSandbox()` and `disableSandbox()` depending on how the Twig env was requested. WDYT?

Reply:
yes, that's a good catch. I registered it as a service now.